TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ultimategoal/vision/GoalFinder.java

```diff
@@ -0,0 +1,165 @@
+package org.firstinspires.ftc.teamcode.ultimategoal.vision;
+
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.opencv.core.*;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.imgproc.Moments;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opencv.imgproc.Imgproc.*;
+
+public class GoalFinder {
+
+    static final double[][] CAMERA_MATRIX = {
+            {1468, 0, 0},
+            {0, 1468, 0},
+            {960, 540, 1}
+    };
+
+    static final Scalar ORANGE = new Scalar(0, 165, 255);
+    static final Scalar BLACK = new Scalar(0, 0, 0);
+
+    public static class GoalLocationData {
```

Comment:
Is nested class good practice?

Reply:
I don't think they're bad practice, just not super common; the reason I did it this way is because the location data imo should be more like a struct or something, and I don't think that immutable records really warrant having their own proper public class